src/tir/transforms/lower_sparse_iter.cc

```diff
@@ -933,6 +933,10 @@ class IterTransformer : public StmtExprMutator {
       body_stmts.push_back(
           BufferStore(mid, BufferLoad(mid, mid_indices) - Integer(1), mid_indices));
     }
+    Stmt then_stmt = BufferStore(mid, -1, mid_indices);
+    PrimExpr if_stmt = (pivot != val);
+    body_stmts.push_back(
+    IfThenElse(if_stmt,then_stmt));
```

Comment:
We might not need to emit this `IfThenElse` if we can confirm the lookup will not fail.